sys/include/net/gnrc/gnrc_mac/packet_queue.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ * @ingroup     net_gnrc_mac
+ * @file
+ * @brief       Wrapper for priority_queue that holds gnrc_pktsnip_t* and is
```

Comment:
> Can you help explain here? I remember, as you have stated somewhere, was it first due to the unawareness of "gnrc_pktqueue " when creating the packet queue module for Lw-MAC

Yes, that's right.

> or does it due to that we want the queue to be aware of its length?

This definetely helps during runtime, but the queues should be quite small anyway.

All in all my packet queue implementation was something I was planning to refactor and maybe merge with the existing one, so in general I'm with @miri64.